kpi/tests/test_assets.py

```diff
@@ -1002,3 +1003,40 @@ def test_asset_is_excluded_from_projects_list_flag(self):
         # Fetch the newly created asset and verify the flag is False
         owner_asset = Asset.objects.get(name='Breakfast')
         self.assertFalse(owner_asset.is_excluded_from_projects_list)
+
+    def test_asset_visibility_after_transfer(self):
+        """
+        Test to ensure that an asset shared with an organization owner remains
+        visible after the asset owner joins a different organization
+        """
+        # Step 1: Create an asset owned by an external user
+        self._create_asset_by_bob()
+        asset = Asset.objects.get(owner=self.external_user)
+
+        # Step 2: External user shares the asset explicitly with OrgA's owner
+        asset.assign_perm(self.org_owner, PERM_VIEW_ASSET)
+        self.assertTrue(self.org_owner.has_perm(PERM_VIEW_ASSET, asset))
+
+        # Step 3: Verify that OrgA's owner can see the asset in their asset list
+        self.client.force_login(self.org_owner)
+        response = self.client.get(self.asset_list_url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(
+            any(obj['uid'] == asset.uid for obj in response.data['results'])
+        )
+
+        # Step 4: Invite the external user to another organization (OrgB)
+        another_org = Organization.objects.create(
+            id='org1234', name='Another Organization', mmo_override=True
+        )
+        another_org.add_user(self.thirduser)
```

Comment:
I guess, `self.thirduser` is the owner of `another_org`, right? 
Would you mind to change the name of the `another_org` and use a more obvious variable name to represent that `thirduser` is the owner. Like `thiruser_org` ;-)